core/src/main/java/org/wildfly/glow/Utils.java

```diff
@@ -405,7 +405,12 @@ public static LayerMapping buildMapping(Map<String, Layer> layers, Set<String> p
                     String[] split = val.split(",");
                     for (String s : split) {
                         s = escapePattern(s);
-                        mapping.getAnnotations().put(s, l);
+                        Set<Layer> ll = mapping.getAnnotations().get(s);
+                        if(ll == null) {
+                            ll = new HashSet<>();
```

Comment:
Just a question, would the order matter?

Also you could do something like:
```java
final Set<Layer> ll = mapping.getAnnotations().computeIfAbsent(s, (value) -> HashSet::new);
```

Then you can skill the `null` check. Not critical at all, just an idea :)

Reply:
@jamezp , excellent suggestion. Done.